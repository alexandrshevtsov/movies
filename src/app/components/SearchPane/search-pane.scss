@import "../../global.scss";

.searchPaneContainer {
  width: calc(100vw - 24px);
  padding: 2px;


  input {
    font-size: 20px;
    line-height: 32px;
    width: calc(100vw - 140px);
    padding-left: 5px;
  }

  button {
    margin-left: 2px;
    font-size: 20px;
    line-height: 32px;
  }
}
