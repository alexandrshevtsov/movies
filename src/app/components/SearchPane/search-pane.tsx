import * as React from 'react';
import styles from './search-pane.scss';

interface ISearchPaneProps {
  onSearch?: (query: string) => void;
  query?: string;
}

export class SearchPane extends React.Component<ISearchPaneProps> {
  inputQuery: HTMLInputElement | null | undefined;

  searchClick = (): void => {
    const onSearch = this.props.onSearch;
    const inputQuery = this.inputQuery;
    if (onSearch && inputQuery) onSearch(inputQuery.value);
  };

  componentDidUpdate() {
    const inputQuery = this.inputQuery;
    if (inputQuery) inputQuery.value = this.props.query || '';
  }

  render() {
    return (
      <div className={styles.searchPaneContainer}>
        <input className={styles.searchText} type="text" maxLength={50} ref={(ref) => (this.inputQuery = ref)} />
        <button className={styles.searchButton} onClick={this.searchClick}>Search</button>
      </div>
    );
  }
}
