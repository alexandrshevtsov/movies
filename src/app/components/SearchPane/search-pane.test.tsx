import * as React from 'react';
import { mount } from 'enzyme';
import { SearchPane } from './search-pane';

describe('SearchPane', () => {
  it('should contain input and button', () => {
    const query = 'black';
    const onSearch = jest.fn();
    const component = mount(<SearchPane query={query} onSearch={onSearch} />);
    const queryText = component.find('input');
    expect(queryText.length).toBe(1);
    const searchButton = component.find('button');
    expect(searchButton.length).toBe(1);
    searchButton.simulate('click');
    expect(onSearch).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
