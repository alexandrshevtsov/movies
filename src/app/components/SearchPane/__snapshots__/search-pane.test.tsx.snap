// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPane should contain input and button 1`] = `
<SearchPane
  onSearch={
    [MockFunction] {
      "calls": Array [
        Array [
          "",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  query="black"
>
  <div>
    <input
      maxLength={50}
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</SearchPane>
`;
